### PR TITLE
(feature) Create usePatient function which updates patient based on URL changes

### PR DESCRIPTION
**Pull request by brandones**
I haven't been able to test this. I link `@openmrs/esm-framework` into a frontend module, but a bunch of typescript build errors come up and my changes aren't reflected in the browser. Leaving off for a minute.

### CODE REVIEW

**Comment by brandones** [comment]
@denniskigen You have a use case for this, yeah? I [can't get](https://github.com/systemjs/systemjs/issues/2379) esm-patient-chart-app to load, and that's the only module I know of that uses it with no `patientUuid` argument. If you're able to test this out, please do.

The approach should be as follows:

```sh
cd openmrs-esm-core/packages/framework/esm-react-utils
yarn link
yarn run:shell
# then in a different terminal
cd whatever-you're-working-on
yarn link @openmrs/esm-react-utils
yarn serve
# add to import map overrides
```

**Comment by denniskigen** [comment]
Hey, @brandones. Just been able to test it out and type errors aside, it appears to work as expected.

**Comment by brandones** [comment]
Awesome! Can you fix the type errors @denniskigen ?

**Comment by jonathandick** [comment]
@denniskigen are you able to help fix the tests?

**Comment by brandones** [comment]
@denniskigen Can I get a review and merge on this?

**Comment by brandones** [comment]
I made it so that this doesn't break anything; it's just a new function. So we can merge it whenever.

**Review by denniskigen — APPROVED** [review]
LGTM! Thanks, @brandones.

File: packages/framework/esm-react-utils/src/usePatient.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  type: ActionTypes.loadError;
+  err: Error | null;
+}
+
+type Action = LoadPatient | NewPatient | PatientLoadError;
+
+enum ActionTypes {
+  loadPatient = "loadPatient",
+  newPatient = "newPatient",
+  loadError = "patientLloadErroroadError",
```
Comment (dkayiwa) [inline packages/framework/esm-react-utils/src/usePatient.ts:33]: Does the loadError have a typo in its value?

**Pull request merged**